### PR TITLE
Add localStorage persistence and retrieval to Try Reason

### LINKED ISSUE #44: Memorize last Try snippet in localStorage

### PR DESCRIPTION
**Pull request by glennsl**
Fixes #44 

Man, this was an absolutely awful coding experience. Why is this not in Reason!? lol

### CODE REVIEW

File: src/pages/try.js [hunk 1]
```diff
@@ -21,32 +21,45 @@ if (typeof navigator !== 'undefined') {
...
+
+const retrieve = () => {
+  function fromQueryParam(language) {
+    const queryParam = window.location.search; // returns ?language=blablabla
+    const prefix = queryParamPrefixFor(language);
+
+    if (queryParam.startsWith(prefix)) {
+      return {
+        language,
+        code: decompress(queryParam.slice(prefix.length)) || '' // decompressing an empty string returns null, joyously!
```
Comment (jaredly) [inline src/pages/try.js]: do you know if decompress ever throws?

File: src/pages/try.js [hunk 2]
```diff
@@ -21,32 +21,45 @@ if (typeof navigator !== 'undefined') {
...
+      };
+    }
   }
-}
 
-const encodeSnippetToURL = debounce((input, queryParamPrefix) => {
-  const encodedSnippet = compress(input);
+  function fromLocalStorage() {
+    const json = localStorage.getItem('try-reason');
+    return json && JSON.parse(json);
```
Comment (jaredly) [inline src/pages/try.js]: this might throw if the localstorage is wrong somehow - could you wrap in a try/catch?

File: src/pages/try.js [hunk 3]
```diff
@@ -21,32 +21,45 @@ if (typeof navigator !== 'undefined') {
...
+  }
+
+  // WTH? There's some retarded automatic semicolon insertion going on, actively causing bugs. Hence the parens. Wonderful!
```
Comment (jaredly) [inline src/pages/try.js]: yup good ol' javascript

File: src/pages/try.js (hunk 2 above)
Comment (glennsl) [reply src/pages/try.js]: Done

File: src/pages/try.js (hunk 1 above)
Comment (glennsl) [reply src/pages/try.js]: It doesn't seem to, it just returns an empty string on invalid input. We could still detect that and default to something more useful though.

File: src/pages/try.js (hunk 3 above)
Comment (glennsl) [reply src/pages/try.js]: I think this is something else. I've written tons of js in this style, with and without babel, and never experienced anything like this.

**Comment by jaredly** [comment]
looks great!

**Comment by chenglou** [comment]
It's not in Reason because we wanted more contributions and we didn't even know whether the new docsite was gonna be stable. We'll rewrite in Reason eventually =)

**Pull request merged**